music21/abcFormat/testFiles.py

```diff
@@ -854,14 +855,15 @@ def testAbc21Chords(self):
             s = translate.abcToStreamScore(ah)
             self.assertEqual(s.duration.quarterLength, quarter_length,
                              'invalid duration of chord "%s"' % abc_chord)
-            for e in s.elements[1].getElementsByClass(chord.Chord):
+            part = s.getElementsByClass(stream.Part)
+            for e in part.getElementsByClass(chord.Chord):
```

Comment:
Similarly here.  I'll bet that `e` is never being set.  Temporarily put a print statement into the loop and you'll see it never runs.

Reply:
I wonder while the stream iterator implements .getElementsByClass at all ?
However .parts[0] looks better.
fixed in the last commit.

Reply:
It totally does.  StreamIterators are designed to be chained, so most explicit cases would be:

```
s.flat.iter.getElementsByOffset(10.0, 20.0).getElementsByClass('Note').getElementsByGroup('cue').addFilter(lambda n: n.pitch.alter == 1)
```

As of v.6., each of these "getElementsBy..." returns a new StreamIterator that adds a new filter on the notes.  So `s.getElementsByClass(stream.Part).getElementsByClass(chord.Chord)` would always return no elements, because first it has to be of class Part, and then of class Chord, and nothing will ever be like that.

It might be that adding a second filter of the same type should raise an exception, but there are a few cases where you might want this, such as:

`s.getElementsByClass(spanner.Spanner).getElementsByClass(expressions.Tremolo)`

which only gets Tremolos that are Spanners. 

Maybe that doesn't work, but there are a few cases like that.